### DISCUSSION TITLE
osiset shopify laravel app configuration

### DISCUSSION THREAD

**Discussion by JIbraeelMehmood** [Kyon147/laravel-shopify#358] [General]
Hi, I'm having trouble connecting my new Laravel 9 project to Shopify using the Osiset package. I'm using an ngrok URL and have set the partner public app Shopify client ID as the API key, with the domain set to store14-test.myshopify.com.

Could you please provide some guidance or steps to configure the Shopify-Laravel app?

Additionally, I remember that back in 2023, a friend of mine created a Laravel app using the Osiset package, but he encountered an issue where the app crashed whenever he navigated to other admin options. After some debugging, he discovered that the authentication tokens were missing when navigating away.

note: if someone share the proper setup details (video/images/zip)  it will be excellent  
Thank you!

**Comment by ismaildasci** [comment]
Hey Luke! 👋

Let’s walk through the steps to configure the Osiset package for connecting your Laravel app to Shopify. I’ll cover the basics and then add a few pointers that might help with the token issue you mentioned.

### Steps to Configure Osiset Shopify Laravel App

1. **Install the Osiset Package**:
   First, install the Osiset package via Composer:
   ```bash
   composer require osiset/laravel-shopify
   ```

2. **Publish and Configure**:
   Publish the config files and views using:
   ```bash
   php artisan vendor:publish --tag=shopify-config
   php artisan vendor:publish --tag=shopify-view
   ```

   Then, update your `.env` file with your Shopify API credentials. These will include your `SHOPIFY_API_KEY` (client ID) and `SHOPIFY_API_SECRET`. Also, configure the `SHOPIFY_APP_URL` to your ngrok URL, as well as other required environment variables:

   ```env
   SHOPIFY_API_KEY=your_shopify_client_id
   SHOPIFY_API_SECRET=your_shopify_client_secret
   SHOPIFY_APP_URL=https://your-ngrok-url.ngrok.io
   SHOPIFY_APP_SCOPES=read_products,write_products
   ```

   Be sure to replace `your_shopify_client_id`, `your_shopify_client_secret`, and `https://your-ngrok-url.ngrok.io` with your actual details.

3. **Set Up Authentication Middleware**:
   In `config/shopify-app.php`, set the `auth_middleware` to `auth.shopify`. This ensures the app uses Shopify's authentication flow.

4. **Migrate Database**:
   Run the migrations to create the necessary tables for storing Shopify data:
   ```bash
   php artisan migrate
   ```

5. **Update App URL in Partner Dashboard**:
   Go to your Shopify Partner Dashboard, and update your app’s URL and redirect URLs to your ngrok URL. For example:
   - **App URL**: `https://your-ngrok-url.ngrok.io`
   - **Redirect URL**: `https://your-ngrok-url.ngrok.io/authenticate`

### Handling Token Issues (Based on Your Friend’s Experience)

To avoid token issues, make sure:
1. **Session Tokens Are Correctly Set Up**: Osiset relies on Shopify's session token. If you notice tokens disappearing, confirm that sessions are managed properly. Laravel should save sessions to a database or a persistent session store.
2. **Use `auth.shopify` Middleware**: This middleware ensures that tokens are checked and refreshed when navigating different routes in the app.
3. **Add Token Middleware to Routes**: Use `auth.shopify` middleware for all routes that require authentication with Shopify.

### Optional: Debugging Tips
- **Log Tokens**: Add logging in your `AuthenticateShopify.php` middleware to check if tokens are being cleared unexpectedly.
- **Check Session Configuration**: In `.env`, set `SESSION_DRIVER` to `database` for better persistence.

Unfortunately, I don’t have a video or zip file, but following these steps should set up the Osiset package correctly. Let me know if you hit any specific issues, and we can debug further! 😊

**Reply by JIbraeelMehmood** [reply]
thanks for your comment
